src/services/WebSocketPolyfill.js

```diff
@@ -83,32 +82,32 @@ export default function initWebSocketPolyfill(syncService, fileId, initialSessio
 			// Useful for debugging what steps are sent and how they were initiated
 			// data.forEach(logStep)
 
-			this.#queue.push(...data)
+			queue.push(...data)
 			let outbox = []
 			return syncService.sendSteps(() => {
-				outbox = [...this.#queue]
+				outbox = [...queue]
 				const data = {
 					steps: this.#steps,
 					awareness: this.#awareness,
 					version: this.#version,
 				}
-				this.#queue = []
+				queue = []
```

Comment:
I don't think this works as expected yet. After a reconnect the first push only contains the initial steps. I suspect that due to assigning a new array the original one in Editor.vue is not adjusted but a local state is introduced instead.

This patch is quite helpful for testing:

(first change avoids the old provider leaking error events, second keeps the editor editable when disconnected to produce some steps)

```
diff --git a/src/components/Editor.vue b/src/components/Editor.vue
index 6b4cce517..cad543a92 100644
--- a/src/components/Editor.vue
+++ b/src/components/Editor.vue
@@ -378,6 +378,8 @@ export default {

                        this.listenSyncServiceEvents()

+                       this.$providers.forEach(p => p?.destroy())
+                       this.$providers = []
                        const syncServiceProvider = createSyncServiceProvider({
                                ydoc: this.$ydoc,
                                syncService: this.$syncService,
@@ -572,7 +574,7 @@ export default {

                onError({ type, data }) {
                        this.$nextTick(() => {
-                               this.$editor?.setEditable(false)
+                               //this.$editor?.setEditable(false)
                                this.emit('sync-service:error')
                        })
```

Reply:
Ahh... good catch! Thanks for testing.
This should be easy to fix.

Reply:
Included the first part in the latest commit as discussed during todays team call.